### PR TITLE
onPan callback added

### PR DESCRIPTION
**Pull request by oguibueno**
added onPan callback to atlas provider so the consumers are able to use it to capture when the map was panned by the user.

### CODE REVIEW

File: packages/atlas/lib/src/provider.dart [hunk 1]
```diff
@@ -29,6 +29,7 @@ abstract class Provider {
     final ArgumentCallback<AtlasController> onMapCreated,
     final ArgumentCallback<CameraPosition> onCameraPositionChanged,
     final ArgumentCallback<LatLng> onLocationChanged,
+    final ArgumentCallback<bool> onPan,
```
Comment (felangel) [inline packages/atlas/lib/src/provider.dart:32]: Thoughts on renaming to onCameraMove and changing to ArgumentCallback<CameraPosition>?

Comment (felangel) [inline packages/atlas/lib/src/provider.dart]: Why is this an `ArgumentCallback<bool>`?

Comment (oguibueno) [reply packages/atlas/lib/src/provider.dart]: I don't need it tbh. I will replace it for a `VoidCallback`, what do you think?

File: packages/atlas/pubspec.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 name: atlas
 description: An extensible map abstraction for Flutter with support for multiple map providers
-version: 0.12.1
+version: 0.13.0
```
Comment (felangel) [inline packages/atlas/pubspec.yaml]:
```suggestion
version: 0.12.2
```

Since this isn't a breaking change, I think it should just be bumped to 0.12.2.

**Review by felangel — CHANGES_REQUESTED** [review]
see comments about versioning also looks like coverage has dropped below 100%. Can this be addressed?

**Comment by oguibueno** [comment]
> see comments about versioning also looks like coverage has dropped below 100%. Can this be addressed?

Do you have any idea why it dropped? These changes aren't directly affecting google_atlas. That's weird

**Comment by felangel** [comment]
> > see comments about versioning also looks like coverage has dropped below 100%. Can this be addressed?
> 
> Do you have any idea why it dropped? These changes aren't directly affecting google_atlas. That's weird

Yeah I agree it seems odd... might be a remnant from a previous PR so we can address it separately 👍

**Comment by felangel** [comment]
Last question, how does `onPan` differ from `onCameraPositionChanged`?

**Comment by oguibueno** [comment]
> Last question, how does `onPan` differ from `onCameraPositionChanged`?

`onPan` is about the gesture and `onCameraPositionChanged` is about the camera state. 

e.g.: 

If you pinch the map (two fingers) to zoom in or zoom out, without panning it, it will only trigger the `onCameraPositionChanged` callback, because the camera state changed. It shouldn't trigger the `onPan` event, since you pinched it or did a double tap.

If you pan the map, it should trigger both, `onCameraPositionChanged` and `onPan` (if it was passed to the Atlas constructor).

The motivation here is to know if was the user doing the pan gesture on the map or if it was the map provider returning all the camera position states.

**Review by felangel — APPROVED** [review]
LGTM ✅

**Comment by oguibueno** [comment]
> LGTM ✅

Thank you!

**Review by edsondiasalves — APPROVED** [review]

**Pull request merged**